api/middleware_test.go

```diff
@@ -0,0 +1,52 @@
+//nolint:noctx,bodyclose
+package api
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTimeoutMiddleware(t *testing.T) {
+	withMiddleware := false
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Wait some time, so that the request timeout (10ms) should be reached.
+		<-time.After(2 * time.Second)
```

Comment:
actually does this work? we don't set up any blocked writes

Reply:
It just tested that the request context was canceled, but now that we use the upstream TimeoutHandler, there's no more need for testing the timeout middleware anymore, so I removed the test.